packages/compose/src/hooks/use-async-list/index.ts

```diff
@@ -26,15 +32,22 @@ function getFirstItemsPresentInState( list, state ) {
 	return firstItems;
 }
 
+type ResetAction< T > = { type: 'reset'; list: T[] };
+type AppendAction< T > = { type: 'append'; item: T };
```

Comment:
`ResetAction` and `AppendAction` are always used in conjunction. Should we create a union type of the two and use that in the 2 instances below?

Reply:
Given @jsnajdr comment I've removed these types completely.